feature-settings-impl/src/main/java/io/novafoundation/nova/feature_settings_impl/presentation/cloudBackup/CloudBackupSettingsViewModel.kt

```diff
@@ -87,34 +100,72 @@ class CloudBackupSettingsViewModel(
     private fun initSyncCloudBackupState() {
         launch {
             if (cloudBackupSettingsInteractor.isSyncCloudBackupEnabled()) {
-                syncBackupInternal(onBackupNotFound = {
-                    /* TODO: run create backup using existing password */
-                })
+                syncBackupInternal(
+                    onSuccess = {},
+                    onBackupNotFound = { /* TODO: run create backup using existing password */ },
+                    onUnknownPassword = {},
+                    onOtherError = {}
+                )
             }
         }
     }
 
+    private fun Throwable.toEnableBackupSyncState(): BackupSyncOutcome {
+        return when (this) {
+            is PasswordNotSaved, is InvalidBackupPasswordError -> BackupSyncOutcome.UnknownPassword
+            is FetchBackupError.BackupNotFound -> BackupSyncOutcome.Ok // not found backup is ok when we enable backup
+            is FetchBackupError.CorruptedBackup -> BackupSyncOutcome.CorruptedBackup
+            is FetchBackupError.Other -> BackupSyncOutcome.UnknownError
```

Comment:
I suggest to use interfaces here: for example we can use CorruptedBackupError instead of FetchBackupError.CorruptedBackup

Reply:
FetchBackupError subclasses already implemented these interfaces

Reply:
In this particular case I think it is clearer to handle all FetchBackupError snce it allows to easier check against what service method throws (see method docs)